src/main.py

```diff
@@ -276,9 +278,14 @@ def main():
         except Exception as e:
             logging.error(e)
             time.sleep(10)
+            
+        if killhandler.killed:
+            logging.info("Termination signal received - exiting")
+            break
 
 
 if __name__ == "__main__":
+    killhandler = KillHandler()
```

Comment:
You'll need to move this instantiation into `main()` as well. And please go ahead and nix the try/catch here as well if you don't mind.